drivers/tty/serial/8250/8250_ts.c

```diff
@@ -52,7 +53,16 @@ static int technologic_ts16550_probe(struct platform_device *pdev)
 	port->serial_in = tsisa_serial_in;
 	port->serial_out = tsisa_serial_out;
 
-	return serial8250_register_8250_port(&uport);
+	line = serial8250_register_8250_port(&uport);
+
+	if (line < 0) {
+		dev_err(dev,
+			"serial8250_register_8250_port() 0x%X irq %d failed\n",
+			port->mapbase, port->irq);
+		return -ENOMEM;
```

Comment:
This should actually `return line;`, as `serial8250_register_8250_port()` returns an error related to what the actual error is.

Reply:
Good point, I had copied someone else's error handling for this function but this makes more sense to return the actual error.  Pushing that change now